src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -208,6 +204,26 @@ internal partial class DefiniteAssignmentPass : LocalDataFlowPass<
             _shouldCheckConverted = this.GetType() == typeof(DefiniteAssignmentPass);
         }
 
+        private static SourceAssemblySymbol? GetSourceAssembly(
+            CSharpCompilation compilation,
+            Symbol member,
+            BoundNode node)
+        {
+            if (member is null)
+            {
+                return null;
+            }
+            if (node.Kind == BoundKind.Attribute)
+            {
+                // member is the attribute type, not the symbol where the attribute is applied.
+                Debug.Assert(member is TypeSymbol type &&
+                    (type.IsErrorType() || compilation.IsAttributeType(type)));
+                return null;
+            }
+            Debug.Assert(member.ContainingAssembly is SourceAssemblySymbol);
```

Comment:
>Debug.Assert(member.ContainingAssembly is SourceAssemblySymbol); [](http://example.com/codeflow?start=12&length=64)

Consider if we can make this assert stronger. Something like:
```
Debug.Assert((object)member.ContainingAssembly == compilation.SourceAssembly); 
```

Reply:
BTW, if we always have ```compilation```, why not simply grab its ```SourceAssembly``` regardless of the ```node``` and ```member```?

Reply:
Reverting to use `member.ContainingAssembly as SourceAssemblySymbol` rather than `compilation.SourceAssembly` since `compilation` may be `null` in the EE. (It doesn't seem like we have any tests that hit this case currently, but it seems safer to allow that.)

Reply:
> since ```compilation``` may be null in the EE.

This doesn't match the nullable annotations on the parameters. Also, we store ```compilation``` in a protected field in the base class and it doesn't look like we protect against `null` value in it.

Reply:
There are cases where `AbstractFlowPass<,>.compilation` is null - see [https://github.com/dotnet/roslyn/issues/59093](https://github.com/dotnet/roslyn/issues/59093).

Logged [https://github.com/dotnet/roslyn/issues/59111](https://github.com/dotnet/roslyn/issues/59111) to ensure callers pass a non-null `CSharpCompilation`.